bids/grabbids/bids_layout.py

```diff
@@ -124,12 +121,60 @@ def get_bval(self, path, **kwargs):
         else:
             return tmp
 
-    def get_events(self, path, **kwargs):
-        tmp = self._get_nearest_helper(path, '.tsv', type='events', **kwargs)
-        if isinstance(tmp, list):
-            return tmp[0]
+    def get_events(self, path, output='file', derivatives='both', **kwargs):
+        """ For a given file in a BIDS project, finds corresponding event files
+        and optionally returns merged dataframe containing all variables.
+
+        Args:
+            path (string): Path to a file
+            output (string): Output to return.
+                'file' returns list of files, 'df' returns merged dataframe.
+            derivatives (string): How to handle derivative events. One of
+                'ignore', 'both', or 'only'
+            how (string): Merge method.
+        Returns:
+            List of file or merged Pandas dataframe.
+        """
+
+        path = abspath(path)
+
+        # Get events in base Layout directory (ordered)
+        root_events = self._get_nearest_helper(
+            path, '.tsv', type='events', **kwargs) or []
+
+        entities = self.files[path].entities
+        if 'type' in entities:
+            entities.pop('type')
+        entities.update(kwargs)
+
+        # Get all events
+        events =  self.get(extensions='tsv', type='events',
+                           return_type='file', **entities) or []
+
+        deriv_events = list(set(events) - set(root_events))
```

Comment:
Derivatives should always have `derivatives/` in the path, so maybe use that to distinguish between the core and deriv files? Seems more elegant than making two separate calls and doing a subtraction